lightningd/jsonrpc.c

```diff
@@ -425,6 +425,14 @@ static void json_command_malformed(struct json_connection *jcon,
 	return json_result(jcon, id, NULL, error);
 }
 
+/* jcon may outlive command. Make sure we don't keep a dangling
+ * pointer from jcon to the cmd */
+static void jcon_disconnect_cmd(struct command *cmd)
+{
+	if (cmd->jcon->current == cmd)
+		cmd->jcon->current = NULL;
+}
+
```

Comment:
Note the symmetry with finish_jcon() which similarly unlinks:

	if (jcon->current) {
		log_unusual(jcon->log, "Abandoning current command");
		jcon->current->jcon = NULL;
	}

So cmd->jcon can be NULL, and your code needs to test this too AFAICT.